### DISCUSSION TITLE
RLS Policies with Database Functions

### DISCUSSION THREAD

**Discussion by davilico05** [supabase/supabase#3571] [Questions] [answered]
Hi!
I'm using "append_array" into database [#2771](https://github.com/supabase/supabase/discussions/2771) and, after a few tests, I need to modify my RLS policies.
How I handle this with RLS policies? Because I have an update rule
`CREATE POLICY "Enable insert for authenticated users only" ON public.comments FOR UPDATE WITH CHECK (auth.role() = 'authenticated');`

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
You need to have USING statement also on update as it reads the record first.  WITH CHECK only looks at incoming replacement data.

**Reply by davilico05** [reply]
Thank you!! It's working!
